rename/rename.py

```diff
@@ -0,0 +1,53 @@
+# Replace all non-filename references to a word (not containing a .filetype) in all files of
+# specified type in a file tree with a new expression
+
+from os import walk
+from os.path import dirname, join, abspath, splitext
+
+
+def replace(dirpath, old_word, new_word, filetypes, changelog_path):
+    changelog = open(changelog_path, "w")
+    dirdata = walk(dirpath)
+
+    for (top_dir, subdirs, file_names) in dirdata.__iter__():
+        for file_name in file_names:
+            if splitext(file_name)[1] in filetypes:
+                filepath = join(top_dir, file_name)
+                changelog.write(filepath + "\n")
+
+                f = open(filepath, "r")
+                lines = []
+                for line in f:
+                    if old_word in line:
+                        line_words = []
+                        for word in line.split(" "):  # type: str
+                            if __valid_word(word, old_word, filetypes):
+                                line_words.append(word.replace(old_word, new_word))
+                            else:
+                                line_words.append(word)
+
+                        lines.append(" ".join(line_words))
+                        changelog.write("\tReplaced: " + line)
+                        changelog.write("\tWith: " + " ".join(line_words))
+                    else:
+                        lines.append(line)
+
+                f.close()
+                f = open(join(top_dir, file_name), "w")
+                f.writelines(lines)
+                f.close()
+    changelog.close()
+
+
+def __valid_word(word, old_word, filetypes):
+    for filetype in filetypes:
+        if filetype in word:
+            return False
+
+    return old_word in word
+
+
+# do
+replace(dirname(abspath(".")), "SEG", "ByteGain", [".h", ".m"], abspath("./changelog.txt"))
```

Comment:
I'm realizing that the problem is more complicated.
Lines that start with "#import" should not be changed.
Occurrences of "SEGMENT" should become "BYTEGAIN".
Remaining occurrences of "SEG" should become "ByteGain".